### DISCUSSION TITLE
Ho to apply policy for authenticated user with condition

### DISCUSSION THREAD

**Discussion by piotrkulpinski** [supabase/supabase#15984] [Questions]
Hi, I have a system where post can be assigned to a specific user, but does not have to. 

So I need to create a RLS policy to the posts table checking if current logged user is the same as user_id or user_id is null. 

I've tried this but it doesn't work for "unclaimed" posts for logged out users: 
`((auth.uid() = user_id) OR ((NULL::uuid = user_id))`

Any ideas? Thanks

**Comment by GaryAustin1** [comment]
You can't test for NULL with = as NULL represents does not exist and not a value.
Use `user_id IS NULL`.
